javascript/frameworks/cap/test/queries/cqlinjection/srv/service1.js

```diff
@@ -0,0 +1,786 @@
+const cds = require("@sap/cds");
+
+module.exports = class Service1 extends cds.ApplicationService {
+  init() {
+    /* ========== 1. Service1 running query on the database service using `cds.run` and friends using Fluent API ========== */
+    this.on("send00111", async (req) => {
+      const { id } = req.data;
+      const query = SELECT.from`Entity1`.where("ID=" + id);
+      cds.run(query);
+    });
+
+    this.on("send00112", async (req) => {
+      const { id } = req.data;
+      const query = SELECT.from`Entity1`.where(`ID=` + id);
+      cds.run(query);
+    });
+
+    this.on("send00113", async (req) => {
+      const { id } = req.data;
+      const query = SELECT.from`Entity1`.where(`ID=${id}`);
+      cds.run(query);
+    });
+
+    this.on("send00114", async (req) => {
+      const { id } = req.data;
+      const query = SELECT.from`Entity1`.where`ID=${id}`;
+      cds.run(query); // FP
+    });
+
+    this.on("send00121", async (req) => {
+      const { id } = req.data;
+      cds.read("Entity1").where("ID =" + id);
+    });
+
+    this.on("send00122", async (req) => {
+      const { id } = req.data;
+      cds.read("Entity1").where(`ID =` + id);
+    });
+
+    this.on("send00123", async (req) => {
+      const { id } = req.data;
+      cds.read("Entity1").where(`ID=${id}`);
+    });
+
+    this.on("send00124", async (req) => {
+      const { id } = req.data;
+      cds.read("Entity1").where`ID=${id}`; // FP
+    });
+
+    this.on("send00131", async (req) => {
+      const { id } = req.data;
+      cds.create("Entity1").entries({id: "" + id});
+    });
+
+    this.on("send00132", async (req) => {
+      const { id } = req.data;
+      cds.create("Entity1").entries({id: `` + id});
+    });
+
+    this.on("send00133", async (req) => {
+      const { id } = req.data;
+      cds.create("Entity1").entries({id: `${id}`});
+    });
+
+    this.on("send00141", async (req) => {
+      const { id, amount } = req.data;
+      cds.update("Entity1").set("col1 = col1" + amount).where("col1 = " + id);
+    });
+
+    this.on("send00142", async (req) => {
+      const { id, amount } = req.data;
+      cds.update("Entity1").set("col1 = col1" + amount).where(`col1 = ` + id);
+    });
+
+    this.on("send00143", async (req) => {
+      const { id, amount } = req.data;
+      cds.update("Entity1").set("col1 = col1" + amount).where(`col1 = ${id}`);
+    });
+
+    this.on("send00144", async (req) => {
+      const { id, amount } = req.data;
+      cds.update("Entity1").set("col1 = col1" + amount).where`col1 = ${id}`; // FP
```

Comment:
Why is this considered an FP? Also, is `set` permitted to have a string argument? The [documentation](https://cap.cloud.sap/docs/node.js/cds-ql#set) says:

> Specifies the data to update...
> 1. As a single-expression tagged template string
> 2. As an object with keys being element names of the target entity and values being simple values, query-by-example expressions, or CQN expressions:

Reply:
Ouch, that's labeled as FP by mistake. The `where` clause is a tagged literal expression `` .where`col1 = ${id}` `` which if it were only by itself would have been an FP (SAFE) case. But the `.set("col = col1" + amount)` sitting there makes it a TP (UNSAFE) case.